test/test-mongocrypt-ctx-encrypt.c

```diff
@@ -2163,6 +2163,373 @@ _test_encrypt_fle2_unindexed_encrypted_payload (_mongocrypt_tester_t *tester)
 }
 #undef RNG_DATA
 
+static mongocrypt_t *
+_crypt_with_rng (_test_rng_data_source *rng_source)
+{
+   mongocrypt_t *crypt;
+   mongocrypt_binary_t *localkey;
+   /* localkey_data is the KEK used to encrypt the keyMaterial
+    * in ./test/data/keys/ */
+   char localkey_data[MONGOCRYPT_KEY_LEN] = {0};
+
+   crypt = mongocrypt_new ();
+   mongocrypt_setopt_log_handler (crypt, _mongocrypt_stdout_log_fn, NULL);
+   localkey = mongocrypt_binary_new_from_data ((uint8_t *) localkey_data,
+                                               sizeof localkey_data);
+   ASSERT_OK (mongocrypt_setopt_kms_provider_local (crypt, localkey), crypt);
+   ASSERT_OK (mongocrypt_setopt_crypto_hooks (
+                 crypt,
+                 _std_hook_native_crypto_aes_256_cbc_encrypt,
+                 _std_hook_native_crypto_aes_256_cbc_decrypt,
+                 _test_rng_source,
+                 _std_hook_native_hmac_sha512,
+                 _std_hook_native_hmac_sha256,
+                 _error_hook_native_sha256,
+                 rng_source /* ctx */),
+              crypt);
+
+   mongocrypt_binary_destroy (localkey);
+   ASSERT_OK (mongocrypt_init (crypt), crypt);
+   return crypt;
+}
+
+static void
+_test_encrypt_fle2_explicit (_mongocrypt_tester_t *tester)
+{
+   _mongocrypt_buffer_t user_key_id;
+   _mongocrypt_buffer_t index_key_id;
+
+   if (!_aes_ctr_is_supported_by_os) {
+      printf ("Common Crypto with no CTR support detected. Skipping.");
+      return;
+   }
+
+   _mongocrypt_buffer_copy_from_hex (&user_key_id,
+                                     "ABCDEFAB123498761234123456789012");
+   _mongocrypt_buffer_copy_from_hex (&index_key_id,
+                                     "12345678123498761234123456789012");
+
+   /* Test Unindexed. */
+   {
+#define RNG_DATA \
+   "\x4d\x06\x95\x64\xf5\xa0\x5e\x9e\x35\x23\xb9\x8f\x57\x5a\xcb\x15"
+      _test_rng_data_source source = {
+         .buf = {.data = (uint8_t *) RNG_DATA, .len = sizeof (RNG_DATA)}};
```

Comment:
`sizeof (RNG_DATA)` includes null terminator. Is this acceptable? This question also applies to other instances of `sizeof (RNG_DATA)` below.

Reply:
Good eye. Preferably no, since the test asserts if more random data was requested than expected. Updated to subtract one.